subsys/bluetooth/services/ots/ots_client.c

```diff
@@ -169,15 +129,89 @@ static int read_attr(struct bt_conn *conn,
 		    struct bt_otc_internal_instance_t *inst,
 		    uint16_t handle, bt_gatt_read_func_t cb);
 
-static void print_oacp_response(enum bt_ots_oacp_proc_type req_opcode,
-				enum bt_ots_oacp_res_code result_code)
+/* L2CAP callbacks */
+static void tx_done(struct bt_gatt_ots_l2cap *l2cap_ctx,
+		    struct bt_conn *conn)
+{
+	/* Not doing any writes yet */
+	BT_ERR("Unexpected call, context: %p, conn: %p", l2cap_ctx, conn);
+}
+
+static ssize_t rx_done(struct bt_gatt_ots_l2cap *l2cap_ctx,
+		       struct bt_conn *conn, struct net_buf *buf)
+{
+	const uint32_t offset = cur_inst->rcvd_size;
+	bool is_complete = false;
+	const struct bt_otc_obj_metadata *cur_object =
+		&cur_inst->otc_inst->cur_object;
+	int cb_ret;
+
+	BT_DBG("Incoming L2CAP data, context: %p, conn: %p, len: %u, offset: %u",
+	       l2cap_ctx, conn, buf->len, offset);
+
+	cur_inst->rcvd_size += buf->len;
+
+	if (cur_inst->rcvd_size >= cur_object->current_size) {
+		is_complete = true;
+	}
+
+	if (cur_inst->rcvd_size > cur_object->current_size) {
+		BT_WARN("Received %u but expected maximum %u",
+			cur_inst->rcvd_size, cur_object->current_size);
+	}
+
+
+	cb_ret = cur_inst->otc_inst->cb->content_cb(conn, offset,
```

Comment:
Probably doesn't need to be done as part of this PR but would be nice to return the number of bytes remaining in this read to stay consistent with the server implementation.

Reply:
https://github.com/zephyrproject-rtos/zephyr/projects/38